src/Catalyst.Core.Modules.Dfs/DfsService.cs

```diff
@@ -0,0 +1,663 @@
+        //             ILogger logger,
+        //             string swarmKey = "07a8e9d0c43400927ab274b7fa443596b71e609bacae47bd958e5cd9f59d6ca3",
+        //             IEnumerable<MultiAddress> seedServers = null)
+        //         {
+        //             if (seedServers == null || seedServers.Count() == 0)
+        //             {
+        //                 seedServers = new[]
+        //                 {
+        //                     new MultiAddress("/ip4/46.101.132.61/tcp/4001/ipfs/18n3naE9kBZoVvgYMV6saMZdtAkDHgs8MDwwhtyLu8JpYitY4Nk8jmyGgQ4Gt3VKNson"),
+        //                     new MultiAddress("/ip4/188.166.13.135/tcp/4001/ipfs/18n3naE9kBZoVvgYMV6saMZe2AAPTCoujCxhJHECaySDEsPrEz9W2u7uo6hAbJhYzhPg"),
+        //                     new MultiAddress("/ip4/167.172.73.132/tcp/4001/ipfs/18n3naE9kBZoVvgYMV6saMZe1E9wXdykR6h3Q9EaQcQc6hdNAXyCTEzoGfcA2wQgCRyg")
+        //                 };
+        //             }
+        //             
+        //             _logger = logger;
+        //
+        //             // The password is used to access the private keys.
+        //             var password = passwordReader.RetrieveOrPromptAndAddPasswordToRegistry(PasswordRegistryTypes.IpfsPassword, "Please provide your IPFS password");
+        //             _ipfs = new Dfs();
+        //             _ipfs.Options.KeyChain.DefaultKeyType = Constants.KeyChainDefaultKeyType;
+        //
+        //             // The IPFS repository is inside the catalyst home folder.
+        //             _ipfs.Options.Repository.Folder = Path.Combine(
+        //                 fileSystem.GetCatalystDataDir().FullName,
+        //                 Constants.DfsDataSubDir);
+        //
+        //             // The seed nodes for the catalyst network.
+        //             _ipfs.Options.Discovery.BootstrapPeers = seedServers;
+        //
+        //             // Do not use the public IPFS network, use a private network
+        //             // of catalyst only nodes.
+        //             _ipfs.Options.Swarm.PrivateNetworkKey = new PreSharedKey
+        //             {
+        //                 Value = swarmKey.ToHexBuffer()
+        //             };
+        //
+        //             _logger.Information("IPFS configured.");
+        //         }
+
+        /// <summary>
+        ///     Determines if the engine has started.
+        /// </summary>
+        /// <value>
+        ///     <b>true</b> if the engine has started; otherwise, <b>false</b>.
+        /// </value>
+        /// <seealso cref="Start" />
+        /// <seealso cref="StartAsync" />
+        public bool IsStarted => _stopTasks.Count > 0;
+
+        private readonly DfsState _dfsState;
+        private readonly SecureString _passphrase;
+        private readonly IHashProvider _hashProvider;
+        private ConcurrentBag<Func<Task>> _stopTasks = new ConcurrentBag<Func<Task>>();
+
+        public DfsService(IBitSwapApi bitSwapApi,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Constructor has 17 parameters, which is greater than the 7 authorized.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)